src/databricks/sql/auth/authenticators.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2022 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"), except
+# that the use of services to which certain application programming
+# interfaces (each, an "API") connect requires that the user first obtain
+# a license for the use of the APIs from Databricks, Inc. ("Databricks"),
+# by creating an account at www.databricks.com and agreeing to either (a)
+# the Community Edition Terms of Service, (b) the Databricks Terms of
+# Service, or (c) another written agreement between Licensee and Databricks
+# for the use of the APIs.
+#
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+from typing import Dict, List
+
+from databricks.sql.auth.oauth import OAuthManager
+import base64
+
+
+# Private API: this is an evolving interface and it will change in the future.
+# Please must not depend on it in your applications.
+from databricks.sql.experimental.oauth_persistence import OAuthToken, OAuthPersistence
+
+
+class CredentialsProvider:
+    def add_headers(self, request_headers: Dict[str, str]):
+        pass
```

Comment:
Ditto previous comment. 
```suggestion
        raise NotImplementedError()
```

Reply:
ditto as above.

This is actually not meant to be an interface but a base class with no op.

what's the use case?
the use case is when the existing `_use_cert_as_auth` and `_tls_client_cert_file` are used, in that scenario we don't add any auth additional header to the request.